suite-native/app/src/StylesProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useColorScheme } from 'react-native';
+
+import { AppColorScheme, selectThemeVariant } from '@suite-native/module-settings';
+import { createRenderer, StylesProvider as StylesStyleProvider } from '@trezor/styles';
+import { prepareNativeTheme, ThemeColorVariant } from '@trezor/theme';
+import { useSelector } from 'react-redux';
+
+type StylesProviderProps = {
+    children: React.ReactNode;
+};
+
+const renderer = createRenderer();
+
+const DEFAULT_COLOR_VARIANT: ThemeColorVariant = 'chill';
+
+const getColorVariant = (
+    systemColorScheme: ReturnType<typeof useColorScheme>,
+    userColorVariant: AppColorScheme,
+) => {
+    if (userColorVariant !== 'system') {
+        return userColorVariant;
+    }
+
+    switch (systemColorScheme) {
+        case 'dark':
+            return 'dark';
+        case 'light':
+            return DEFAULT_COLOR_VARIANT;
+        default:
+            return DEFAULT_COLOR_VARIANT;
+    }
+};
+
+export const StylesProvider = ({ children }: StylesProviderProps) => {
+    const systemColorScheme = useColorScheme();
+    const userThemeVariant = useSelector(selectThemeVariant);
```

Comment:
just a nit when we are talking about conventions: why are these derived data prefixed with _user_? I would expect something like useSelector(selectUserThemeVariant) but it is really a nit and I am more curious than pushing something.

Reply:
Because it's variant picked by user, nothing deep. I can change it but it's good enough and make things little bit obvious in my opinion.